src/colmap/scene/reconstruction.h

```diff
@@ -266,28 +266,46 @@ size_t Reconstruction::NumRegImages() const { return reg_image_ids_.size(); }
 size_t Reconstruction::NumPoints3D() const { return points3D_.size(); }
 
 const struct Camera& Reconstruction::Camera(const camera_t camera_id) const {
-  return cameras_.at(camera_id);
+  try {
+    return cameras_.at(camera_id);
+  } catch (const std::out_of_range& e) {
+    throw std::invalid_argument(
+        StringPrintf("Camera with ID %d does not exist", camera_id));
+  }
 }
 
 const class Image& Reconstruction::Image(const image_t image_id) const {
-  return images_.at(image_id);
+  try {
+    return images_.at(image_id);
+  } catch (const std::out_of_range& e) {
+    throw std::invalid_argument(
+        StringPrintf("Image with ID %d does not exist", image_id));
+  }
 }
 
 const struct Point3D& Reconstruction::Point3D(
     const point3D_t point3D_id) const {
-  return points3D_.at(point3D_id);
+  try {
+    return points3D_.at(point3D_id);
+  } catch (const std::out_of_range& e) {
+    throw std::invalid_argument(
+        StringPrintf("Point3D with ID %d does not exist", point3D_id));
+  }
 }
 
 struct Camera& Reconstruction::Camera(const camera_t camera_id) {
-  return cameras_.at(camera_id);
+  return const_cast<struct Camera&>(
```

Comment:
I'd suggest replicating the code here instead of the const cast. I am not sure if the compiler makes some optimizations with the const that then get violated by this const cast. Code replication is not all that bad and here the savings are also marginal.

Reply:
Ok, done.